File: frontend/src/types/campaign.ts
export interface CampaignCreateRequest {
  name: string;
  description?: string;
}

export interface CampaignJoinRequest {
  inviteCode: string;
}

export interface CampaignMember {
  userId: string;
  username: string;
  displayName: string;
  role: 'DM' | 'PLAYER';
  joinedAt: string;
}

export interface Campaign {
  id: string;
  name: string;
  description?: string;
  dmUserId: string;
  dmDisplayName: string;
  inviteCode: string;
  isActive: boolean;
  members: CampaignMember[];
  createdAt: string;
}

File: backend/src/main/java/com/questkeeper/campaign/Campaign.java
package com.questkeeper.campaign;

import com.questkeeper.user.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "campaigns")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Campaign {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(length = 200, nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dm_user_id", nullable = false)
    private User dm;

    @Column(name = "invite_code", length = 8, unique = true, nullable = false)
    private String inviteCode;

    @Column(name = "is_active")
    @Builder.Default
    private Boolean isActive = true;

    @OneToMany(mappedBy = "campaign", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<CampaignMember> members = new ArrayList<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Instant updatedAt;
}
